server/src/main/java/org/candlepin/resource/PoolResource.java

```diff
@@ -267,6 +268,29 @@ public void deletePool(@PathParam("pool_id") String id) {
     }
 
     /**
+     * Retrieve a CDN for a Pool
+     *
+     * @return the cdn for a pool
+     * @httpcode 400
+     * @httpcode 200
+     */
+    @GET
+    @Path("{pool_id}/cdn")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Cdn getPoolCdn(
+        @PathParam("pool_id") @Verify(Pool.class) String id) {
+
+        Pool pool = poolManager.find(id);
+
+        if (pool == null) {
+            throw new NotFoundException(i18n.tr(
+                "Subscription Pool with ID ''{0}'' could not be found.", id));
+        }
+
+        return pool.getCdn();
```

Comment:
@Ceiu I thought about it, and I think today we essentially respond with a null on a pool object, and since there are no explicit requirements around this API, nor do we know any existing user , I thought it best to maintain that legacy behavior and not error out.
That being said, one could use the same arguments to argue otherwise ( consistency ).

I dont personally have any preference, as this isnt a feature used yet, cause I it might change if its ever used anyway, so I'll defer to any preference you have. thoughts?